linode/manager
# feat: [UIE-8089] - DBaaS Resize GA

## Description 📝
DBaaS Resize GA

## Changes  🔄
List any change relevant to the reviewer.
- Added Number of Nodes selector to the Database Resize tab
- The node selection is cleared if they have 2 nodes option selected and try to switch to a plans tab where it is unavailable.
- Updated newly added Node selector so that the plan is cleared if 2 nodes option is selected for a plan that does not have it.
- Updated summary so that both current and resize cluster pricing summaries are displayed for comparison

## Target release date 🗓️
10/14/24

## Preview 📷

| Before  | After   |
| ------- | ------- |
| ![Before](https://github.com/user-attachments/assets/e8b04cfc-4826-4157-a393-2cf5c120ea30)| ![After](https://github.com/user-attachments/assets/e10f84bf-4fb6-48eb-872d-6bcad414d1af)|

## How to test 🧪

### Prerequisites
(How to setup test environment)
- For testing this change, ensure that DBaaS v2 is enabled, Beta is turned off, and the user has the "Managed Databases" account capability

### Verification steps
(How to verify changes)
- Navigate to the Databases section.
- Select the New Databases tab and click into one of the databases to see your details
- Navigate to the Resize tab
- Scroll down below the plans selection to view the Number of Nodes section is visible
- Verify that the correct node options are visible in the different tabs (ie. Dedicated vs Shared, etc)
- Verify that your current cluster size is selected and that lower sizes are disabled
- Verify that making a higher node selection enables the Resize button
- Verify that updated summary is visible and updates with selection
- Make an invalid selection for the 2 node option to see the summary remove the current selection

## As an Author I have considered 🤔

*Check all that apply*
